vehicle_kinematic_state_node:
  ros__parameters:
    gps_publish_s: 0.05
    imu_publish_s: 0.005
    late_threshold_sec: 0.2
    publish_tf: true
    num_gps_subscribers: 2
    num_imu_subscribers: 2
    min_lat_thresh: 0.35
    min_lon_thresh: 0.35
    max_cov0_thresh: 330.0
    max_cov7_thresh: 330.0
    max_jerk_thresh: 300.0
    max_acc_thresh: 40.0
    max_yaw_rate: 4.0
    heading_cutoff_freq: 10.0
    speed_cutoff_freq: 10.0
    imu_lin_cutoff: 1.0
    imu_ang_cutoff: 1.5
    target_frame: "cg"
    publish_odom: true
global_extended_kalman_filter_node:
  ros__parameters:
    # use_sim_time: true
    frequency: 100.0
    sensor_timeout: 0.005
    two_d_mode: true
    transform_time_offset: 0.0
    transform_timeout: 0.0
    print_diagnostics: false
    debug: false
    # debug_out_file: /path/to/debug/file.txt
    publish_tf: true
    publish_acceleration: true
    reset_on_time_jump: true
    predict_to_current_time: true
    use_control: false

    map_frame: map                   # Defaults to "map" if unspecified
    odom_frame: odom                 # Defaults to "odom" if unspecified
    base_link_frame: base_link # Defaults to "base_link" if unspecified
    world_frame: odom                # Defaults to the value of odom_frame if unspecified

    #        [x_pos   , y_pos    , z_pos,
    #         roll    , pitch    , yaw,
    #         x_vel   , y_vel    , z_vel,
    #         roll_vel, pitch_vel, yaw_vel,
    #         x_accel , y_accel  , z_accel]

    ############################################
    odom0: odom0
    odom0_config: [ true, true, false,
                    false, false, true,
                    true, false, false,
                    false, false, false,
                    false, false, false]
   
    odom0_queue_size: 10

    ############################################
    odom1: odom1
    odom1_config: [ false, false, false,
                    false, false, false,
                    true, false, false,
                    false, false, true,
                    false, false, false]
    odom1_queue_size: 10

    odom0_pose_rejection_threshold: 5.0

    # ############################################
    # imu0: imu0
    # imu0_config: [ false, false, false,
    #                 false, false, false,
    #                 false, false, false,
    #                 false, false, true,
    #                 false, false, false]
    # imu0_queue_size: 10
    
    # ############################################
    # imu1: imu1
    # imu1_config: [ false, false, false,
    #                false, false, false,
    #                false, false, false,
    #                false, false, true,
    #                false, false, false]
    # imu1_queue_size: 10
